### PR TITLE
Adjust XML format

### PR DESCRIPTION
**Pull request by yzqzss**
see: https://github.com/mediawiki-client-tools/mediawiki-scraper/pull/57

code not tested on py2.

### CODE REVIEW

File: dumpgenerator.py [hunk 1]
```diff
@@ -1090,7 +1090,7 @@ def makeXmlFromPage(page):
                 ))
             else:
                 revision.append(E.contributor(deleted="deleted"))
-            if 'comment' in rev:
+            if 'comment' in rev and rev["comment"]:
```
Comment (nemobis) [inline dumpgenerator.py:1093]: Is this what special:export does?

Comment (yzqzss) [reply dumpgenerator.py:1093]: Yes, `special:export` and `dumpBackup.php` (*MW CLI*) both do it this way.  
You can verify this on [these test sites](https://github.com/mediawiki-client-tools/mediawiki-scraper/issues/7#issuecomment-1374490611).

**Comment by nemobis** [comment]
@Pokechu22 Are you able to test this?

**Review by Pokechu22 — APPROVED** [review]
Tested on http://group2.mediawiki.demo.save-web.org/mediawiki-1.31.16/index.php?title=%E9%A6%96%E9%A1%B5 (as that has exactly 1 page with 1 revision). Here's the output: [464.zip](https://github.com/WikiTeam/wikiteam/files/11593939/464.zip). `--xml` didn't change, but `--xml --xmlrevisions` did:

```diff
diff --git a/xmlrevisions_old.xml b/xmlrevisions_new.xml
index c0b943f..cbe18c2 100644
--- a/xmlrevisions_old.xml
+++ b/xmlrevisions_new.xml
@@ -37,7 +37,7 @@
   <revision>
     <id>1</id>
     <timestamp>2023-01-07T12:33:44Z</timestamp>
-    <text bytes="724" space="preserve">&lt;strong&gt;已安装MediaWiki。&lt;/strong&gt;
+    <text bytes="724" xml:space="preserve">&lt;strong&gt;已安装MediaWiki。&lt;/strong&gt;
 
 请查阅[https://www.mediawiki.org/wiki/Special:MyLanguage/Help:Contents 用户指导]以获取使用本wiki软件的信息！
 
@@ -51,7 +51,6 @@
       <username>MediaWiki default</username>
       <id>0</id>
     </contributor>
-    <comment></comment>
     <model>wikitext</model>
     <parentid>0</parentid>
     <sha1>d9b7ee8b27b108d6a4c6b4630283e3df8fbad568</sha1>
```

The output does more closely match that from Special:Export. There are still whitespace and tag order differences but I don't think those have any semantic meaning (unlike an empty tag).

**Comment by nemobis** [comment]
Thank you both! I'm very happy that someone other than me is finally testing and fixing this code path.

**Pull request merged**